core/src/main/kotlin/me/kcra/takenaka/core/mapping/resolve/impl/QuiltMappingResolver.kt

```diff
@@ -0,0 +1,269 @@
+import me.kcra.takenaka.core.mapping.MappingContributor
+import me.kcra.takenaka.core.mapping.matchers.isConstructor
+import me.kcra.takenaka.core.mapping.resolve.AbstractMappingResolver
+import me.kcra.takenaka.core.mapping.resolve.LicenseResolver
+import me.kcra.takenaka.core.mapping.resolve.Output
+import me.kcra.takenaka.core.mapping.resolve.lazyOutput
+import me.kcra.takenaka.core.mapping.util.unwrap
+import me.kcra.takenaka.core.util.*
+import mu.KotlinLogging
+import net.fabricmc.mappingio.MappingReader
+import net.fabricmc.mappingio.MappingUtil
+import net.fabricmc.mappingio.MappingVisitor
+import net.fabricmc.mappingio.adapter.MappingNsRenamer
+import net.fabricmc.mappingio.tree.MappingTree
+import java.net.URL
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.StandardCopyOption
+import java.util.zip.ZipFile
+import kotlin.io.path.bufferedReader
+import kotlin.io.path.fileSize
+import kotlin.io.path.isRegularFile
+import kotlin.io.path.reader
+
+private val logger = KotlinLogging.logger {}
+
+/**
+ * A resolver for the Quilt Mappings from QuiltMC.
+ *
+ * @property workspace the workspace
+ * @property quiltProvider the Quilt metadata provider
+ * @property relaxedCache whether output cache verification constraints should be relaxed
+ * @author Matouš Kučera
+ * @author Florentin Schleuß
+ */
+class QuiltMappingResolver(
+    override val workspace: VersionedWorkspace,
+    val quiltProvider: QuiltMetadataProvider,
+    val relaxedCache: Boolean = true
+) : AbstractMappingResolver(), MappingContributor, LicenseResolver {
+    override val licenseSource: String
+        get() = "https://raw.githubusercontent.com/QuiltMC/quilt-mappings/${version.id}/LICENSE"
+    override val targetNamespace: String = "quilt"
+    override val outputs: List<Output<out Path?>>
+        get() = listOf(mappingOutput, licenseOutput)
+
+    /**
+     * Creates a new resolver with a default metadata provider.
+     *
+     * @param workspace the workspace
+     * @param xmlMapper an [ObjectMapper] that can deserialize XML trees
+     * @param relaxedCache whether output cache verification constraints should be relaxed
+     */
+    constructor(workspace: VersionedWorkspace, xmlMapper: ObjectMapper, relaxedCache: Boolean = true)
+            : this(workspace, QuiltMetadataProvider(workspace, xmlMapper), relaxedCache)
+
+    override val mappingOutput = lazyOutput<Path?> {
+        resolver {
+            val file = workspace[MAPPING_JAR]
+
+            val builds = quiltProvider.versions[version.id]
+            if (builds == null) {
+                logger.info { "did not find Quilt Mappings mappings for ${version.id}" }
+                return@resolver null
+            }
+
+            val targetBuild = builds.maxBy(QuiltBuild::buildNumber)
+
+            var urlString = "https://maven.quiltmc.org/repository/release/org/quiltmc/quilt-mappings/$targetBuild/quilt-mappings-$targetBuild-intermediary-mergedv2.jar"
```

Comment:
Quilt seems to do better than Fabric and has mergedv2 for all of their versions, so we can simplify this to just non-intermediary mergedv2 (and a no classifier artifact as a fallback, which seems to be the same as mergedv2 for some reason?)